File: src/java/foundgine-core/src/main/java/com/foundgine/core/semantic/metadata/IMetadataProvider.java
package com.foundgine.core.semantic.metadata;

import com.foundgine.core.abstractions.*;

public interface IMetadataProvider {
	EntityMetadata getEntity(EntityId entityId);

	RelationshipMetadata getRelationship(RelationshipId relationshipId);

	ModelMetadata getModel(ModelId modelId);

	ConnectionMetadata getConnection(ConnectionId connectionId);

	ConversionMetadata findConversion(Class<?> sourceType, Class<?> targetType);

	AuthorizationMetadata getAuthorization(AuthorizationId authorizationId);
}

File: src/csharp/Foundgine.Core/Semantic/Metadata/IMetadataProvider.cs
using Foundgine.Core.Abstractions;

namespace Foundgine.Core.Semantic.Metadata;

/// <summary>
/// Runtime access to generated/static metadata. The core depends on this
/// abstraction rather than on a particular generator implementation.
/// </summary>
public interface IMetadataProvider
{
    EntityMetadata GetEntity(EntityId entityId);
    RelationshipMetadata GetRelationship(RelationshipId relationshipId);
    ModelMetadata GetModel(ModelId modelId);
    ConnectionMetadata GetConnection(ConnectionId connectionId);
    ConversionMetadata? FindConversion(Type sourceType, Type targetType);
    AuthorizationMetadata GetAuthorization(AuthorizationId authorizationId);
}
